### DISCUSSION TITLE
Env file

### DISCUSSION THREAD

**Discussion by albertopasqualetto** [tgdrive/teldrive#192] [Q&A]
Hello, it is not clear if an instance running in docker needs the `.env` file even if `config.toml` is present.

I am asking since I get `ERROR   file streamretry middleware skip: rpcDoRequest: rpc error code 400: CHANNEL_INVALID` when I upload something.

**Comment by divyam234** [comment]
you need to set default channel from UI first its should be a private channel on telegram

**Reply by albertopasqualetto** [reply]
I already did that and I see it correctly in the database

**Reply by PLoliwi** [reply]
I also have the following problems 

19/04/2024 04:25 PM	ERROR	errorretry middleware skip: rpcDoRequest: rpc error code 400: CHANNEL_INVALID
19/04/2024 04:25 PM	ERROR	callback: retry middleware skip: rpcDoRequest: rpc error code 400: CHANNEL_INVALID

**Reply by PLoliwi** [reply]
@albertopasqualetto Did you make it?

**Reply by PLoliwi** [reply]
Chrome/124.0.0.0 Safari/537.36", "latency": 0.000082196, "time": "2024-04-19T16:30:59Z"}
19/04/2024 04:31 PM	ERROR	[DB] github.com/divyam234/teldrive/pkg/services/user.go:117 ERROR: null value in column "channel_id" of relation "channels" violates not-null constraint (SQLSTATE 23502)
[74.029ms] [rows:0] INSERT INTO "teldrive"."channels" ("channel_name","user_id","selected") VALUES ('',6130007053,true) ON CONFLICT ("channel_id") DO UPDATE SET "selected"=true RETURNING "channel_id"
19/04/2024 04:31 PM	ERROR	failed to update channel

**Reply by divyam234** [reply]
Repull latest build this is fixed now

**Reply by PLoliwi** [reply]
@divyam234 

runtime error: index out of range [1] with length 1
goroutine 247 [running]:
net/http.(*conn).serve.func1()
	net/http/server.go:1898 +0xbe
panic({0x1972a60?, 0xc0000567b0?})
	runtime/panic.go:770 +0x132
github.com/divyam234/teldrive/internal/category.GetCategory({0xc000740520?, 0xc0000f8300?})
	github.com/divyam234/teldrive/internal/category/category.go:28 +0x2a5
github.com/divyam234/teldrive/pkg/services.(*FileService).CreateFile(0xc00045a3a8, 0xc0000f8300, 0x16d607c0d, 0xc000032d80)
	github.com/divyam234/teldrive/pkg/services/file.go:82 +0x3d8
github.com/divyam234/teldrive/pkg/controller.(*Controller).CreateFile(0xc0000b8460, 0xc0000f8300)
	github.com/divyam234/teldrive/pkg/controller/file.go:27 +0x145
github.com/gin-gonic/gin.(*Context).Next(0xc0000f8300)
	github.com/gin-gonic/gin@v1.9.1/context.go:174 +0x2b
github.com/divyam234/teldrive/api.InitRouter.Authmiddleware.func1(0xc0000f8300)
	github.com/divyam234/teldrive/internal/middleware/middleware.go:80 +0x294
github.com/gin-gonic/gin.(*Context).Next(...)
	github.com/gin-gonic/gin@v1.9.1/context.go:174
github.com/divyam234/teldrive/cmd.initApp.func1(0xc0000f8300)
	github.com/divyam234/teldrive/cmd/run.go:226 +0x1a3
github.com/gin-gonic/gin.(*Context).Next(0xc0000f8300)
	github.com/gin-gonic/gin@v1.9.1/context.go:174 +0x2b
github.com/divyam234/teldrive/cmd.initApp.GinzapWithConfig.func4(0xc0000f8300)
	github.com/gin-contrib/zap@v1.1.1/zap.go:68 +0xb1
github.com/gin-gonic/gin.(*Context).Next(...)
	github.com/gin-gonic/gin@v1.9.1/context.go:174
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc00036e820, 0xc0000f8300)
	github.com/gin-gonic/gin@v1.9.1/gin.go:620 +0x66e
github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc00036e820, {0x216e938, 0xc000132460}, 0xc000426000)
	github.com/gin-gonic/gin@v1.9.1/gin.go:576 +0x1b2
net/http.serverHandler.ServeHTTP({0x216a750?}, {0x216e938?, 0xc000132460?}, 0x6?)
	net/http/server.go:3137 +0x8e
net/http.(*conn).serve(0xc00067e120, {0x2171518, 0xc0003b6a50})
	net/http/server.go:2039 +0x5e8
created by net/http.(*Server).Serve in goroutine 51
	net/http/server.go:3285 +0x4b4
	
	
	you messed something up
	
	
![image](https://github.com/divyam234/teldrive/assets/95148122/91e88cbf-51ab-48d1-9a7e-6725e08183bd)

**Reply by albertopasqualetto** [reply]
@PLoliwi issue happens also to me, but with status code 502

> runtime error: index out of range [1] with length 1
goroutine 59 [running]:
net/http.(*conn).serve.func1()
        net/http/server.go:1898 +0xb0
panic({0x14e9540?, 0x40019f8168?})
        runtime/panic.go:770 +0x124
github.com/divyam234/teldrive/internal/category.GetCategory({0x4001d0a570?, 0x400062a200?})
        github.com/divyam234/teldrive/internal/category/category.go:28 +0x2e8
github.com/divyam234/teldrive/pkg/services.(*FileService).CreateFile(0x4000524168, 0x400062a200, 0x8bc89c7, 0x4000596300)
        github.com/divyam234/teldrive/pkg/services/file.go:82 +0x384
github.com/divyam234/teldrive/pkg/controller.(*Controller).CreateFile(0x40001bc4a0, 0x400062a200)
        github.com/divyam234/teldrive/pkg/controller/file.go:27 +0x100
github.com/gin-gonic/gin.(*Context).Next(0x400062a200)
        github.com/gin-gonic/gin@v1.9.1/context.go:174 +0x40
github.com/divyam234/teldrive/api.InitRouter.Authmiddleware.func1(0x400062a200)
        github.com/divyam234/teldrive/internal/middleware/middleware.go:80 +0x23c
github.com/gin-gonic/gin.(*Context).Next(...)
        github.com/gin-gonic/gin@v1.9.1/context.go:174
github.com/divyam234/teldrive/cmd.initApp.func1(0x400062a200)
        github.com/divyam234/teldrive/cmd/run.go:226 +0x190
github.com/gin-gonic/gin.(*Context).Next(0x400062a200)
        github.com/gin-gonic/gin@v1.9.1/context.go:174 +0x40
github.com/divyam234/teldrive/cmd.initApp.GinzapWithConfig.func4(0x400062a200)
        github.com/gin-contrib/zap@v1.1.1/zap.go:68 +0x88
github.com/gin-gonic/gin.(*Context).Next(...)
        github.com/gin-gonic/gin@v1.9.1/context.go:174
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0x400052c4e0, 0x400062a200)
        github.com/gin-gonic/gin@v1.9.1/gin.go:620 +0x518
github.com/gin-gonic/gin.(*Engine).ServeHTTP(0x400052c4e0, {0x1ce6898, 0x4001c36000}, 0x400063c240)
        github.com/gin-gonic/gin@v1.9.1/gin.go:576 +0x174
net/http.serverHandler.ServeHTTP({0x1ce26b0?}, {0x1ce6898?, 0x4001c36000?}, 0x6?)
        net/http/server.go:3137 +0xbc
net/http.(*conn).serve(0x40000be6c0, {0x1ce94e8, 0x4000589170})
        net/http/server.go:2039 +0x508
created by net/http.(*Server).Serve in goroutine 16
        net/http/server.go:3285 +0x3f0

**Reply by albertopasqualetto** [reply]
But I also notice that files are uploaded to the channel

**Reply by PLoliwi** [reply]
Good build  @albertopasqualetto

**Reply by albertopasqualetto** [reply]
@PLoliwi ???
